modernize MolViewer

speleo3 opened the pull request:
- use `subprocess` instead of `os.system` thread
- automatically start headless server
- `stop()` method
- stop server if instance goes out of scope
- display: wait for image, show progress bar

Minimal example to show a ray traced image:
```python
from ipymol import viewer as pymol
pymol.fetch('3odu')
pymol.display()
```

speleo3 committed:
modernize MolViewer

- use subprocess instead of os.system thread
- automatically start headless server
- stop() method
- stop server if instance goes out of scope
- display: wait for image, show progress bar

cxhernandez commented on ipymol/core.py:
```diff
@@ -16,54 +15,93 @@ class MolViewer(object):
+            self._server.png(filename, 0, 0, -1, int(ray))
+
+            # ~2min timeout
+            for i in range(1, 50):
```
Might be nice to define a parameter `timeout` that a user can set, since rendering time can vary a lot depending on hardware and render settings.

speleo3 replied:
`timeout` added

cxhernandez commented on ipymol/core.py:
```diff
@@ -16,54 +15,93 @@ class MolViewer(object):
+                time.sleep(i / 10.0)
+
+            return Image(filename)
+        finally:
```
My suggestion here would be to add a bit more error handling and warnings so that a user knows whether the `timeout` was exceeded or if the image could not be read/displayed.

speleo3 replied:
I added a timeout parameter and an exception if the timeout exceeded.

cxhernandez commented on ipymol/core.py:
```diff
@@ -16,54 +15,93 @@ class MolViewer(object):
+                progress.value += 1
+                time.sleep(i / 10.0)
+
+            return Image(filename)
```
Since we're using `IPython.display.Image` now, it would be nice to expose the `width` and `height` parameters in `display`.

speleo3 replied:
Makes sense. I added width and height.

cxhernandez reviewed (approved):
Thanks @speleo3! This looks great! 

I have a few nits about exposing more configurable parameters and error handling, but I'm happy to take those on myself in a separate PR.

speleo3 committed:
display(width, height, ray, timeout)

- add width/height/timeout parameters to display function
- raise exception on timeout

speleo3 commented:
thanks @cxhernandez for merging!
